helix-core/src/main/java/org/apache/helix/messaging/handling/HelixTaskExecutor.java

```diff
@@ -99,14 +100,23 @@ public MsgHandlerFactoryRegistryItem(MessageHandlerFactory factory, int threadPo
         throw new IllegalArgumentException("Illegal thread pool size: " + threadPoolSize);
       }
 
+      if (resetTimeout <= 0) {
```

Comment:
Nit: 0 probably should be allowed. If 0 is not allowed, why is 1 (or other small numbers) allowed?

Reply:
Agreed. Will change.